src/stepFunctions/asl/asl-yaml-languageservice.ts

```diff
@@ -0,0 +1,249 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* eslint-disable no-useless-escape */
+/* eslint-disable @typescript-eslint/unbound-method*/
+
+import {
+    ASL_SCHEMA,
+    doCompleteAsl,
+    ClientCapabilities,
+    getLanguageService as getASLLanguageService,
+    LanguageService,
+} from 'amazon-states-language-service'
+import {
+    DocumentLanguageSettings,
+    JSONDocument,
+    getLanguageService as getLanguageServiceVscode,
+    LanguageServiceParams,
+} from 'vscode-json-languageservice'
+import {
+    parse as parseYAML,
+    YAMLDocument,
+    SingleYAMLDocument,
+} from 'yaml-language-server/out/server/src/languageservice/parser/yamlParser07'
+import {
+    CompletionItem,
+    CompletionList,
+    DocumentSymbol,
+    FormattingOptions,
+    Hover,
+    TextEdit,
+    TextDocument,
+    Position,
+    Range,
+    SymbolInformation,
+} from 'vscode-languageserver-types'
+import { matchOffsetToDocument } from 'yaml-language-server/out/server/src/languageservice/utils/arrUtils'
+import { YAMLSchemaService } from 'yaml-language-server/out/server/src/languageservice/services/yamlSchemaService'
+import { YAMLCompletion } from 'yaml-language-server/out/server/src/languageservice/services/yamlCompletion'
+import * as URL from 'url'
+import * as prettier from 'prettier'
+
+const workspaceContext = {
+    resolveRelativePath: (relativePath: string, resource: string) => {
+        return URL.resolve(resource, relativePath)
+    },
+}
+
+const aslLanguageService = getASLLanguageService({
+    workspaceContext,
+    contributions: [],
+    clientCapabilities: ClientCapabilities.LATEST,
+    aslOptions: {
+        ignoreColonOffset: true,
+    },
+})
+
+const requestServiceMock = function(uri: string): Promise<string> {
```

Comment:
The requestServiceMock is essentially an empty parameter for the yaml-language-server to initialize a schemaService. The design is based off of how the yaml-language-server tests auto-completion using a mock schemaRequest service. It would be ideal to pin the version of the yaml-language-server dependency.